rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -98,8 +99,14 @@ private void doRecoverTransaction(Uid recoverUid) {
             boolean inFlight = (lra.getLRAStatus() == LRAStatus.Active);
 
             LRAStatus lraStatus = lra.getLRAStatus();
-            if (LRAStatus.FailedToCancel.equals(lraStatus) || LRAStatus.FailedToClose.equals(lraStatus)) {
-                return;
+
+            synchronized (this) {
```

Comment:
synchronized here, is necessary?

Reply:
I remove it from here as well. I thought the same scenario could happen here as well, but this would not help much.